### ISSUE TITLE
Chaining test

### ISSUE THREAD

**Issue by chrisvidal** [dareid/chakram#64]
I know from the chakram tests itself that we can chained requests which is great.
Now i would like to chain tests with promise
my scenario is the following one:
- get all folders and get the total count
- post a new folder and test the response
- get all folders and test that the count is now +1
- get the folder details from the last id and test the content of data
- update the details and test the response
- delete the folder and test the response
- finally get all the folders and test the count is back to initial value

basically chained request and chaining test.

Is this process possible ?

thanks for any ideas
